packages/ckeditor5-image/src/imageinsert/imageinsertviaurlui.ts

```diff
@@ -77,62 +85,30 @@ export default class ImageInsertViaUrlUI extends Plugin {
 		// Set initial value because integrations are created on first dropdown open.
 		imageInsertUrlView.imageURLInputValue = replaceImageSourceCommand.value || '';
 
-		this._imageInsertUI.dropdownView!.on( 'change:isOpen', () => {
-			if ( this._imageInsertUI.dropdownView!.isOpen ) {
-				// Make sure that each time the panel shows up, the URL field remains in sync with the value of
-				// the command. If the user typed in the input, then canceled and re-opened it without changing
-				// the value of the media command (e.g. because they didn't change the selection), they would see
-				// the old value instead of the actual value of the command.
-				imageInsertUrlView.imageURLInputValue = replaceImageSourceCommand.value || '';
-
-				if ( collapsibleView ) {
-					collapsibleView.isCollapsed = true;
-				}
-			}
-
-			// Note: Use the low priority to make sure the following listener starts working after the
-			// default action of the drop-down is executed (i.e. the panel showed up). Otherwise, the
-			// invisible form/input cannot be focused/selected.
-		}, { priority: 'low' } );
-
-		imageInsertUrlView.on<ImageInsertUrlViewSubmitEvent>( 'submit', () => {
-			if ( replaceImageSourceCommand.isEnabled ) {
-				editor.execute( 'replaceImageSource', { source: imageInsertUrlView.imageURLInputValue } );
-			} else {
-				editor.execute( 'insertImage', { source: imageInsertUrlView.imageURLInputValue } );
-			}
-
-			this._closePanel();
-		} );
-
-		imageInsertUrlView.on<ImageInsertUrlViewCancelEvent>( 'cancel', () => this._closePanel() );
-
-		if ( collapsibleView ) {
-			collapsibleView.set( {
-				isCollapsed: true
-			} );
-
-			collapsibleView.bind( 'label' ).to( this._imageInsertUI, 'isImageSelected', isImageSelected => isImageSelected ?
-				t( 'Update image URL' ) :
-				t( 'Insert image via URL' )
-			);
-
-			return collapsibleView;
-		}
-
 		return imageInsertUrlView;
 	}
 
 	/**
 	 * Creates the toolbar button.
 	 */
-	private _createInsertUrlButton( isOnlyOne: boolean ): ButtonView {
-		const ButtonClass = isOnlyOne ? DropdownButtonView : ButtonView;
+	private _createInsertUrlButton<T extends typeof ButtonView | typeof MenuBarMenuListItemButtonView>(
+		ButtonClass: T
+	): InstanceType<T> {
+		// const ButtonClass = isOnlyOne ? DropdownButtonView : ButtonView;
 
 		const editor = this.editor;
-		const button = new ButtonClass( editor.locale );
+		const button = new ButtonClass( editor.locale ) as InstanceType<T>;
 		const t = editor.locale.t;
 
+		button.set( {
+			label: t( 'Set image url' ),
+			icon: icons.imageUpload
+		} );
```

Comment:
Clean this up. We use `.set()` below as well, which overwrites the icon. Also, the label does not seem correct (and we bind it below, but that should be kept only for toolbar).